app/Anime.php

```diff
@@ -5,9 +5,12 @@
 use App\Http\HttpHelper;
 use Jikan\Jikan;
 use Jikan\Request\Anime\AnimeRequest;
+use Illuminate\Database\Eloquent\Factories\HasFactory;
 
 class Anime extends JikanApiSearchableModel
 {
+    use HasFactory;
+
     // note that here we skip "score", "min_score", "max_score", "rating" and others because they need special logic
     // to set the correct filtering on the ORM.
     protected array $filters = ["order_by", "status", "type", "sort"];
```

Comment:
> note that here we skip "score", "min_score", "max_score", "rating" and others because they need special logic to set the correct filtering on the ORM.

Bit of a newbie on this approach. How will these filters be handled in the testing? And what's the purpose of this `filters` property here?

Reply:
Handling these filters: we just provide them during tests.
What's the purpose of the `filters` property: See [`FilterQueryString.php` ](https://github.com/jikan-me/jikan-rest/blob/master/app/Filters/FilterQueryString.php#L44) [There are scopes in laravel eloquent](https://laravel.com/docs/9.x/eloquent#query-scopes), and I tried to use that to add a `->filter()` function on all query builder instances. This "scope" method is responsible for mapping the query string parameters to filter instructions.